Modifications to the structure and the content of the introduction (#139)
SORTEE-Github-Hackathon/manuscript: content/02.introduction.md

## Introduction {.page_break_before}

### General introduction to the use of collaborative computational resources by scientists

*Contributors to this section: PHPB*

Most scientists, including ecologists and evolutionary biologists, have become critically dependent on computational tools in their research [see @doi:10.1109/SECSE.2009.5069155].

> Researchers now write and use code as part of their research workflow to perform a wide-variety of tasks facilitating scientific elaboration, ranging from data management, data analysis, study replication, to the application and the development of tools for hypothesis testing.

Reviewer: Should this maybe include some mention of reproducibility as a motivation and/or benefit for using code?

To facilitate this process, scientists have been increasingly borrowing and adopting tools from  information system technology, such as cloud-based services for documentation and version control [_e.g._ from the Google Suite (with Docs, Sheets and Drive), the Microsoft Suite (with Word, Excel and OneDrive), and Github] [@doi: 10.1038/538127a].
However, most researchers lack exposure to adequate software development practices and are required to dedicate valuable time and effort to self-teach the use of research-faciliting tools, and thus may find practical barriers when applying adequate standards in the maintenance of their scientific code [@doi:10.1145/2063348.2063374; @doi:10.1371/journal.pbio.1001745; @doi:10.1109/SECSE.2009.5069155].
Here, we review and discuss one of the most used web-based platforms for computational version control and collaboration, Github, and provide researchers in ecology and evolutionary biology with practical workflows aiming at facilitating their scientific code and management process.

### General background about GitHub and Git

*Contributors to this section: RCO, SSHS, PHPB*

> With over 73 million registered users, GitHub became the most common web-platform used for documentation and collaboration on computer code [@https://github.com/about].

Reviewer: ```suggestion
With over 73 million registered users, GitHub is the most common web-platform used for documentation and collaboration on computer code [@https://github.com/about].
```

Author: Done! Thank you, @Aariq!

> [GitHub](https://github.com/) combines [Git](https://git-scm.com/), its underlying version control system, to provide less complicated, yet powerful services allowing users to participate in projects, contribute, report and discuss code issues, and search for existing code as well as publish their own code.

Reviewer: combines Git with what? Maybe something like... 
```suggestion
[GitHub](https://github.com/) builds on the [Git](https://git-scm.com/) version control system, providing a simplified but powerful web interface that allows users to participate in projects, contribute code, report and discuss software bugs, discover existing code and data, as well as publishing their own.
```

Author: Great contributions! I just incorporated them! Thanks again!

Through its version control, users have a detailed, chronological record on the files and directories stored in their repositories [@doi:10.1080/00031305.2017.1399928].

> This workflow provides a strong and clear advantage over receiving, processing and sending files back-and-forth, a process that can easily become challenging and time-consuming in projects extending in time and in the number of collaborators [@doi:10.1186/1751-0473-8-7].

Reviewer: I think maybe somewhere around here we need a sentence explaining that Github (and git) are tools built with software developers in mind, but that many of the benefits of these tools apply to EEB researchers.

Author: This information was incorporated in the following paragraph, which begins on line 34, with "_The voluminous user-community and the numerous resources..._". 

This paragraph addresses the lack of domain-specific resources for ecology and evolutionary biology researchers. 

The following sentence (in line 36) discusses the point you mention: 
> "Because GitHub and its features have been created centered on collaboration for software development in information systems [@doi:10.1109/ASONAM.2016.7752419], first-time users from domains without formal training in information technology can face challenging learning curves."

### General background on how GitHub is used by EEB researchers

*Contributors to this section: RCO, PHPB*

The voluminous user-community and the numerous resources providing pedagogical instruction material on how to use GitHub streamlined its widespread use (CITE).
Nevertheless, although researchers in ecology and evolutionary biology (EEB) have been encouraged to adopt GitHub in part of their research process [@doi:10.1038/s41559-017-0160; @doi:10.1038/538127a], its use is still not widespread.
Because GitHub and its features have been created centered on collaboration for software development in information systems [@doi:10.1109/ASONAM.2016.7752419], first-time users from domains without formal training in information technology can face challenging learning curves.
Morever, domain-specific perspectives and resources providing tractable examples and practical guidance for researchers in ecology and evolution on GitHub are scarce (but see CITE, CITE, CITE).
An increased availability of data and code managenent standards -- of which GitHub is one increasingly important component -- make research more reproducible and collaborative [@doi:10.1002/bes2.1801].

> More importantly, a widespread, common adoption of GitHub for collaborating on a variety of research tasks can ultimately enable EEB researchers to spend less time on creating novel processes for collaboration and more time on their scientific research [@doi:10.3897/rio.6.e56508].

Reviewer: This is great!

### Our objective

*Contributors to this section: RCO, PHPB*

This manuscript is the result from an academic hackathon held during the 2021 conference for the [Society for Open, Reliable, and Transparent Ecology and Evolutionary Biology (SORTEE)](https://www.sortee.org/).
We convened a group of ~30 EEB researchers with varying levels of familiarity with using GitHub as part of their research projects to showcase and discuss how existing features can contribute to the documentation and collaboration in ecological and evolutionary research.
During the hackathon, we identified the need for a formal discussion on how EEB researchers can benefit from GitHub and its features to make their research more collaborative and transparent.

> We outline thirteen practical workflows on GitHub features that can help towards a more collaborative, transparent and reproducible EEB research.

Reviewer: ```suggestion
Here, we outline thirteen practical uses of (?) GitHub features that can help towards a more collaborative, transparent and reproducible EEB research.
```

Author: Incorporated! Thank you for the recommendation!
